Poker.BE/Poker.BE.Domain/Security/UserManager.cs

```diff
@@ -44,22 +50,28 @@ public class UserManager
             return false;
         }
 
-        public User LogIn (string userName, string password){
-            if (CheckExistingUser(userName)){ // We check that the user is existing in our DB
-               User UserToCheck;
-               if (UsersDictionary.TryGetValue(userName, out UserToCheck) ){ // We take the User Object from our DB
-                   string GoodPassword = UserToCheck.Password;
-                   bool arePasswordMatching = GoodPassword.Equals(password, StringComparison.Ordinal); // We check that the password is correct
-                   if (arePasswordMatching) {
-                       UserToCheck.Connect();
-                       return UserToCheck;
-                    }
-               }
-            }
-            return null;
-        }
+		public User LogIn(string userName, string password)
+		{
+			if (!CheckExistingUser(userName)) // We check that the user is existing in our DB
+				throw new UserNotFoundException();
+			User UserToCheck;
+			if (UsersDictionary.TryGetValue(userName, out UserToCheck))
+			{ // We take the User Object from our DB
+				string GoodPassword = UserToCheck.Password;
+				bool arePasswordMatching = GoodPassword.Equals(password, StringComparison.Ordinal); // We check that the password is correct
```

Comment:
i meant - brake this to 2 lines too...
```c#
bool arePasswordMatching = GoodPassword.Equals(password, StringComparison.Ordinal); // We check that the password is correct
```

![image](https://cloud.githubusercontent.com/assets/9889268/26532232/dfb0459a-4403-11e7-90a0-0dd8a98e1c33.png)

anyway, this file will be override. no worries. don't invest anymore time in it.